keras_nlp/layers/token_and_position_embedding.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Layer that combines an Embedding Layer and Keras-NLP Positional Embedding Layer"""
+
+from tensorflow import keras
+
+import keras_nlp.layers
+
+
+class TokenAndPositionEmbedding(keras.layers.Layer):
+    """Creates a layer which creates Embedding for tokens and adds Position Embedding
+    to the input sequence
+
+    This class assumes that in the input tensor, the last dimension corresponds
+    to the features, and the dimension before the last corresponds to the
+    sequence.
+
+    Args:
+        `vocabulary_size`: The size of the vocabulary (should be no larger than 999)
+        `max_length`: The maximum length of input sequence
+        `embedding_dim`: The output dimension of the embedding layer
+        `embeddings_initializer`: The initializer to use for the Embedding Layer
+        `embeddings_regularizer`: The regularizer to user for regularization of the Embedding Layer
+        `mask_zero`: Boolean, whether or not the input value 0 is a special "padding" value that should be masked out.
+            This is useful when using recurrent layers which may take variable length input. If this is True, then all
+            subsequent layers in the model need to support masking or an exception will be raised. If mask_zero is set
+            to True, as a consequence, index 0 cannot be used in the vocabulary
+            (input_dim should equal size of vocabulary + 1).
+
+    Example:
+    ```python
+    token_and_position_embed = keras_nlp.layers.TokenAndPositionEmbedding(
+        vocabulary_size=vocab_size,
+        max_length=max_length,
+        embedding_dim=embed_dim,
+    )
+    outputs = token_and_position_embed(inputs)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        max_length,
+        embedding_dim,
+        embeddings_initializer="glorot_uniform",
+        embeddings_regularizer=None,
+        mask_zero=False,
+        **kwargs
+    ):
+        super().__init__()
+        if vocabulary_size is None:
+            raise ValueError(
+                "`vocabulary_size` must be an Integer, not `None`."

```

Comment:
We use "received" as a key word in many of our error messages to denote user input. So replace "not `None`" -> "received `None`."

Reply:
I was following `PositionEmbedding` class [ValueError message](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/layers/position_embedding.py#L67). Should we refactor that too (small change in this PR)?